components/feature/readerview/src/main/java/mozilla/components/feature/readerview/ReaderViewFeature.kt

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.readerview
+
+import android.support.annotation.VisibleForTesting
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.webextension.WebExtension
+import mozilla.components.concept.toolbar.Toolbar
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.log.logger.Logger
+
+class ReaderViewFeature(
+    private val engine: Engine,
+    private val toolbar: Toolbar,
```

Comment:
In the latest Fenix mocks "Reader View" is not a page action anymore but instead in this little drawer above the URL bar. So I assume we will need some kind of callback here so that the app can do something custom. I was actually wondering whether we should just expose this as a field on `Session`? This way we could have two feature classes, one installs the extension, updates the Session etc. And one the deals with the page action (which Fenix wouldn't need/use though!).

Reply:
Good point, this wasn't thought out yet on my part. I've added a TODO to change to a callback (instead of passing in the toolbar), then the app can decide how to handle it. The idea of 2 feature classes sounds good as well, but I would like to try it out once we have messaging. Let's  chat more!